icu4c/source/common/loclikely.cpp

```diff
@@ -1291,7 +1297,15 @@ ulocimp_minimizeSubtags(const char* localeID,
                         UErrorCode* status) {
     char localeBuffer[ULOC_FULLNAME_CAPACITY];
 
-    if (do_canonicalize(localeID, localeBuffer, sizeof localeBuffer, status)) {
+    int32_t canonicalizedSize = do_canonicalize(localeID, localeBuffer, sizeof localeBuffer, status);
+    if (*status == U_BUFFER_OVERFLOW_ERROR || *status == U_STRING_NOT_TERMINATED_WARNING) {
+        icu::LocalMemory<char> heapLocaleBuffer((char*)uprv_malloc(canonicalizedSize + 2));
```

Comment:
same as above, check the nullptr of malloc and set U_MEMORY_ALLOCATION_ERROR if nullptr

Reply:
Yes, I caught this one too.